src/Controller/Component/SecurityComponent.php

```diff
@@ -281,18 +318,83 @@ protected function _validatePost(Controller $controller)
         if (empty($controller->request->data)) {
             return true;
         }
+        $token = $this->_validToken($controller);
+        $hashParts = $this->_hashParts($controller);
+        $check = Security::hash(implode('', $hashParts), 'sha1');
+
+        if ($token === $check) {
+            return true;
+        }
+
+        if (Configure::read('debug')) {
+            $msg = $this->_debugPostTokenNotMatching($controller, $hashParts);
+            throw new SecurityException($msg);
+        }
+
+        return false;
```

Comment:
Shouldn't this throw an exception on failure? The caller doesn't check this return value and the new code assumes that no exception == pass.